src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -562,9 +646,17 @@ private Conversion ClassifyStandardImplicitConversion(BoundExpression sourceExpr
         }
 
         private Conversion ClassifyStandardImplicitConversion(TypeSymbol source, TypeSymbol destination, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+        {
+            var conversion = ClassifyStandardImplicitConversionInternal(source, destination, ref useSiteInfo);
+            Debug.Assert(conversion.Kind == ConversionKind.NoConversion || IsStandardImplicitConversionFromType(conversion.Kind));
+            return conversion;
+        }
+
+        private Conversion ClassifyStandardImplicitConversionInternal(TypeSymbol source, TypeSymbol destination, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
         {
             Debug.Assert((object)source != null);
             Debug.Assert((object)destination != null);
+            Debug.Assert(source is not FunctionTypeSymbol);
```

Comment:
>Debug.Assert(source is not FunctionTypeSymbol); [](http://example.com/codeflow?start=12&length=47)

It feels like it would be better to handle this input rather than asserting that it doesn't occur.